File: 프로그래머스/1/42862. 체육복/체육복.js
function solution(n, lost, reserve) {
    // 1. 도난 + 여벌 겹치는 학생 처리 (본인 분만 채우고, 빌려줄 수 없음)
    const realLost = lost.filter(x => !reserve.includes(x));
    const realReserve = reserve.filter(x => !lost.includes(x));

    // 2. 기본: n - 도난당한 인원 수
    let answer = n - realLost.length;

    // 3. realLost 오름차순 정렬 후, 앞·뒤 여벌 체육복 확인
    realLost.sort((a, b) => a - b);
    const reserveSet = new Set(realReserve);

    for (const student of realLost) {
        if (reserveSet.has(student - 1)) {  // 왼쪽 학생이 여벌
            reserveSet.delete(student - 1);
            answer++;
        } else if (reserveSet.has(student + 1)) {  // 오른쪽 학생이 여벌
            reserveSet.delete(student + 1);
            answer++;
        }
    }

    return answer;
}

File: 프로그래머스/1/42862. 체육복/체육복.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        // 1. 도난 + 여벌 겹치는 학생 제거
        Set<Integer> lostSet = new HashSet<>();
        Set<Integer> reserveSet = new HashSet<>();

        for (int l : lost)      lostSet.add(l);
        for (int r : reserve)   reserveSet.add(r);

        // 겹치는 학생은 둘 다 제거 (본인만 입음, 빌려줄 수 없음)
        Set<Integer> temp = new HashSet<>(lostSet);
        for (int l : temp) {
            if (reserveSet.contains(l)) {
                lostSet.remove(l);
                reserveSet.remove(l);
            }
        }

        // 2. 기본 듣는 학생 수
        int answer = n - lostSet.size();

        // 3. lostSet 오름차순 순회 (정렬)
        int[] lostArray = lostSet.stream().mapToInt(i -> i).toArray();
        java.util.Arrays.sort(lostArray);

        for (int student : lostArray) {
            if (reserveSet.contains(student - 1)) {  // 왼쪽 여벌
                reserveSet.remove(student - 1);
                answer++;
            } else if (reserveSet.contains(student + 1)) {  // 오른쪽 여벌
                reserveSet.remove(student + 1);
                answer++;
            }
        }

        return answer;
    }
}